book/website/reproducible-research/code-reuse/code-reuse.md

```diff
@@ -0,0 +1,78 @@
+
+When we talk about making code reusable, it is useful to clarify what is we mean. In the {ref}`Table of Definitions for Reproducibility<rr-overview-definitions-reproducibility>` we defined reproducible research as using the same data and the same code. However, when we talk about code re-use this can take many forms: we may want to run the exact same code (for compiled programming languages, this could even mean the exact same binary file), or we may want to modify the source code and extend it in some particular way to fit our needs. Freire and Chirigati {cite:ps}`Freire2018Reproducibility` provide a framework of different levels of reproducibility, depending on what can be modified. They define the following levels of reproducibility: repeatable, re-runnable, portable, extendable and modifiable.
+
+We can map the definitions of reproducibly on the Freire framework as follows:
+
+| Freire framework | Definitions of reproducibly |
+|------------------|---|
+| Repeatable       | Reproducible (same data, same analysis) |
+| Re-runnable      | Robust & Replicable (same code, different data/analysis/parameters) |
+| Portable         | *Not considered* (Same code/data, different environment) |
+| Extendable       | (partly) Generalisable |
+| Modifiable       | (partly) Generalisable |
+
+Portability was not previously considered, but for software a different environment (hardware, operating system, etc) may affect the ability for the software to work (e.g. it may affect dependencies).
+
+Also, Generalisable encapsulates two concepts: Extendable (the ability to integrate with other software)
+and Modifiable (the ability to change part of the implementation to extend its functionality).
+
+In the rest of this chapter we provide list of recommendations you can follow to make sure your code is reusable.
+
+## How to make your code more reusable
+This section contains a checklist of recommendations for making your software more reusable. The {ref}`rr-code-reuse-recommendation-details` section contains a more in-depth explanation of each of these recommendations. You can follow the recommendations that are more suitable for your type of software and skip the ones which are not relevant in your case.
+
+### Repeatable recommendations
+1. Make sure you can find it (in space)
+1. Make sure you can find it (in time)
+1. Make sure you can execute the same sequence of operations
+1. Make sure your environment and sequence of operations is robust and no human is needed to replicate what was done
+1. License your code
+  - with a license that allows for reuse;
+  - with a license compatible with the dependencies’ licenses
```

Comment:
The indentation is broken
![image](https://user-images.githubusercontent.com/1068752/141147810-ff49f8e4-d764-47b8-80a2-c486061ece3f.png)
Maybe the following:
```suggestion
1. License your code
    - with a license that allows for reuse;
    - with a license compatible with the dependencies’ licenses
```